### PR TITLE
[FIX-#4084][server]fix taskInstance state change error

### LINKED ISSUE #4084: taskInstance state change error
phenomenon：
taskInstance state change from 7 to 1，and user observes the state change from "SUCCESS" to "RUNNING"。
master still loop for check taskInstance state，wait for taskInstance's state is finish.
reason：
master receive the message of TaskResponseProcessor first，and than receive the TaskAckProcessor。

### PR DESCRIPTION
**Pull request by CalvinKirs**
Concurrent processing of ack message and result message causes the execution sequence to be wrong

Optimize code and reduce database queries

this close #4084

### CODE REVIEW

File: dolphinscheduler-server/src/main/java/org/apache/dolphinscheduler/server/master/processor/queue/TaskResponseService.java [hunk 1]
```diff
@@ -126,51 +127,52 @@ public void run() {
...
                 try {
                     TaskInstance taskInstance = processService.findTaskInstanceById(taskResponseEvent.getTaskInstanceId());
-                    if (taskInstance != null){
-                        processService.changeTaskState(taskResponseEvent.getState(),
-                                taskResponseEvent.getStartTime(),
-                                taskResponseEvent.getWorkerAddress(),
-                                taskResponseEvent.getExecutePath(),
-                                taskResponseEvent.getLogPath(),
-                                taskResponseEvent.getTaskInstanceId());
+                    if (taskInstance != null && ExecutionStatus.SUCCESS.getCode() != taskInstance.getState().getCode()) {
```
Comment (lenboo) [inline dolphinscheduler-server/src/main/java/org/apache/dolphinscheduler/server/master/processor/queue/TaskResponseService.java:141]: i think we should judge the task status finished: success/failed/stop...

Comment (CalvinKirs) [reply dolphinscheduler-server/src/main/java/org/apache/dolphinscheduler/server/master/processor/queue/TaskResponseService.java]: Thx,I have corrected it

**Review by lenboo — APPROVED** [review]
+1

**Review by lgcareer — APPROVED** [review]
+1

**Review by dailidong — APPROVED** [review]
+1

**Pull request merged**